src/coloring/greedy_star1_coloring.jl

```diff
@@ -0,0 +1,67 @@
+"""
+        greedy_star1_coloring
+
+    Find a coloring of a given input graph such that
+    no two vertices connected by an edge have the same
+    color using greedy approach. The number of colors
+    used may be equal or greater than the chromatic
+    number χ(G) of the graph.
+
+    A star coloring is a special type of distance - 1  coloring,
+    For a coloring to be called a star coloring, it must satisfy
+    two conditions:
+
+    1. every pair of adjacent vertices receives distinct  colors
+    (a distance-1 coloring)
+
+    2. For any vertex v, any color that leads to a two-colored path
+    involving v and three other vertices  is  impermissible  for  v.
+    In other words, every path on four vertices uses at least three
+    colors.
+
+    reference: What Color is your Jacobian?, pg 662
+"""
+function greedy_star1_coloring(G::VSafeGraph)
```

Comment:
given that you are not using the vertex-safety property, we could relax the type constraint to any graph, `g::LightGraphs.AbstractGraph`